csharp/ql/src/Likely Bugs/LeapYear/UnsafeYearConstruction.ql

```diff
@@ -0,0 +1,35 @@
+/**
+ * @name Unsafe year argument for DateTime constructor
+ * @description Constructing a DateTime object by setting the year argument by manipulating the year of a different DateTime object
+ * @kind problem
+ * @problem.severity error
+ * @id cs/leap-year/unsafe-year-contruction
+ * @precision high
+ * @tags security
+ *       leap-year
+ */
+
+import csharp
+import semmle.code.csharp.dataflow.TaintTracking
+
+class UnsafeYearCreationFromArithmeticConfiguration extends TaintTracking::Configuration  {
+  UnsafeYearCreationFromArithmeticConfiguration() { this = "UnsafeYearCreationFromArithmeticConfiguration" }
+
+  override predicate isSource(DataFlow::Node source) { 
+    exists( ArithmeticOperation ao, PropertyAccess pa |
+      ao = source.asExpr() | 
+      pa = ao.getAChild*()
+      and pa.getProperty().getQualifiedName().matches("%DateTime.Year")
+    )
+  }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists( ObjectCreation oc |
```

Comment:
Indentation here is a bit non-standard: I recommend running the QL autoformatter (pressing `CTRL+SHIFT+F` in QL4E).